### PR TITLE
Use ursadb's iterators in mquery

### PR DESCRIPTION
**Pull request by msm-code**
Additionaly, fix things that broke the CI again.
TODO: move the CI pipeline to github...

Reviewable, but should not be merged until ursadb's iterators go through the review.

### CODE REVIEW

File: src/daemon.py [hunk 1]
```diff
@@ -57,62 +55,101 @@ def collect_expired_jobs() -> None:
...
-        queue, data = redis.blpop(
-            ["queue-search", "queue-index", "queue-metadata", "queue-commands"]
-            + yara_queues
-        )
-
-        if queue == "queue-search":
-            job_hash = data
-            logging.info("New task: {}:{}".format(queue, job_hash))
+        if try_to_do_task(not did_anything):
+            logging.info("Task done successfully...")
```
Comment (ITAYC0HEN) [inline src/daemon.py:139]: Story of my life...

File: src/lib/ursadb.py [hunk 2]
```diff
@@ -25,9 +25,11 @@ def query(self, query: str, taint: str) -> Json:
         start = time.clock()
         if taint:
             taint = taint.replace('"', '\\"')
-            query = 'select with taints ["{}"] {};'.format(taint, query)
+            query = 'select with taints ["{}"] into iterator {};'.format(
```
Comment (nazywam) [inline src/lib/ursadb.py:28]: Any reason for using formats instead of fstrings like in rest of the code?

File: src/daemon.py [hunk 3]
```diff
@@ -148,107 +170,77 @@ def update_metadata(job_hash: str, file_path: str, matches: List[str]) -> None:
...
+    rule = compile_yara(job_hash)
 
-        pipe = redis.pipeline()
-        pipe.rpush("false_positives:" + job_hash, file)
-        pipe.hget("job:{}".format(job_hash), "total_files")
-        pipe.hincrby("job:{}".format(job_hash), "files_processed")
-        _, total_files, files_processed = pipe.execute()
+    for sample in files:
+        try:
+            matches = rule.match(data=open(sample, "rb").read())
```
Comment (nazywam) [inline src/daemon.py:194]: I guess the FD gets closed after the yara matching right?

btw: the `match` function accepts filepaths as well so no need to read the data ourselves right?

Comment (nazywam) [reply src/daemon.py:194]: ref: https://yara.readthedocs.io/en/v3.4.0/yarapython.html#yara.Rules

File: src/lib/ursadb.py (hunk 2 above)
Comment (msm-code) [reply src/lib/ursadb.py:28]: No, it was just like this before my changes. Fixed!

File: src/daemon.py (hunk 3 above)
Comment (msm-code) [reply src/daemon.py:194]: Thanks! I didn't notice this, and I think this is unnecessarily slowing down scanning.

**Pull request merged**